parquet-hadoop/src/main/java/parquet/hadoop/mapred/DeprecatedParquetInputFormat.java

```diff
@@ -39,11 +41,15 @@
   @Override
   public RecordReader<Void, Container<V>> getRecordReader(InputSplit split, JobConf job,
                   Reporter reporter) throws IOException {
-    return new RecordReaderWrapper<V>(realInputFormat, split, job, reporter);
+    return new RecordReaderWrapper<V>(split, job, reporter);
   }
 
   @Override
   public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
+    if (isTaskSideMetaData(job)) {
+      return toParquetSplits(super.getSplits(job, numSplits));
+    }
+
```

Comment:
This method reads the file footers before delegating to `getSplits(Configuration, List<Footer>)` to avoid needing to construct a mapreduce `JobContext` object. But now that we only want to read footers if we are using client-side metadata, this needs to either 1) delegate to `ParquetInputFormat` earlier or 2) know when to read or not read the footers.

Delegating earlier requrires constructing the `JobContext` mapreduce object, while knowing whether or not to read the footer requires `isTaskSideMetadata` (using `ParquetInputFormat.TASK_SIDE_METADATA`). I think the second option is cleaner so that's what I did here.